### ISSUE TITLE
Can't localise inherited ZFS properties because of over-eager impotency

### ISSUE THREAD

**Issue by mavit** [ansible/ansible-modules-extras#1092]
##### Issue Type:
- Feature Idea
##### Ansible Version:

```
ansible 1.9.3
  configured module search path = None
```
##### Environment:

Solaris 11.
##### Summary:

Many ZFS properties are inheritable from the parent filesystem.

Ansible does not distinguish between local, default, and inherited values.  This makes it impossible to set the local value of a property when the inherited value is already as desired.

It would be possible to modify the module so the user can choose whether they want to ignore inherited values or not, but this is probably overkill; if the user was happy with the inherited value, they probably wouldn't have added a task to their play to set it.  If this seems reasonable, I can work on a patch.
##### Steps To Reproduce:

``` yaml
- zfs: state=present setuid=on  name=rpool/export/home/trusteduser 
- zfs: state=present setuid=off name=rpool/export/home
```
##### Expected Results:

```
> zfs get -r setuid rpool/export
NAME                           PROPERTY  VALUE  SOURCE
rpool/export                   setuid    on     default
rpool/export/home              setuid    off    local
rpool/export/home/trusteduser  setuid    on     local
rpool/export/home/otheruser    setuid    off    inherited from rpool/export/home
```
##### Actual Results:

```
> zfs get -r setuid rpool/export
NAME                           PROPERTY  VALUE  SOURCE
rpool/export                   setuid    on     default
rpool/export/home              setuid    off    local
rpool/export/home/trusteduser  setuid    off    inherited from rpool/export/home
rpool/export/home/otheruser    setuid    off    inherited from rpool/export/home
```

**Comment by robynbergeron** [comment]
Hi @mavit -- thanks for submitting this feature idea.

@johanwiren -- pinging you to let you know about this feature request for the zfs module. It looks like @mavit might be willing to implement the feature himself and submit a PR -- if you could give him some feedback on his suggestions, that would be great.

Thanks! :)

**Comment by robynbergeron** [comment]
[module: system/zfs.py]

**Comment by johanwiren** [comment]
Isn't this the exact same behaviour you get when running the zfs command in the same order? In that case I'd say that the current behaviour is what would give you the least amount of surprises. I currently do not have a zfs box at hand to verify this myself. Will try to find one soon though.

Also, reading the zfs man-page I cannot find a way to specifically set a property as "local" if its value at the same time matches that of the inherited one.

Can you elaborate on how you would implement this? Or post a PR if you have working code.

**Comment by johanwiren** [comment]
ping @mavit

**Comment by mavit** [comment]
If you set a value for a property, that value is by definition the local value for that property.  The inherited value is only used if no local value is set.  Hope I've explained that clearly.

**Comment by mavit** [comment]
So, my proposal is: when checking the current value of a property, if source != "local" then value is considered to be null.

**Comment by johanwiren** [comment]
Reproduced it. Will try to work on a fix.

**Comment by johanwiren** [comment]
The fix is included in #1378
